pkg/tenant/members.go

```diff
@@ -261,6 +262,99 @@ func (s *Server) MemberUpdate(c *gin.Context) {
 	c.JSON(http.StatusOK, m.ToAPI())
 }
 
+func (s *Server) MemberRoleUpdate(c *gin.Context) {
+	var err error
+
+	// Members exist on organizations
+	// This method handles the logging and error responses
+	var orgID ulid.ULID
+	if orgID = orgIDFromContext(c); ulids.IsZero(orgID) {
+		return
+	}
+
+	// Get the member ID from the URL and return a 400 if the member ID is not a ULID.
+	var memberID ulid.ULID
+	if memberID, err = ulid.Parse(c.Param("memberID")); err != nil {
+		sentry.Warn(c).Err(err).Str("id", c.Param("memberID")).Msg("could not parse member id")
+		c.JSON(http.StatusNotFound, api.ErrorResponse("member not found"))
+		return
+	}
+
+	// TODO: Add org verification
+
+	// Bind the user request with JSON.
+	params := &api.UpdateMemberParams{}
+	if err = c.BindJSON(&params); err != nil {
+		sentry.Warn(c).Err(err).Msg("could not parse member update request")
+		c.JSON(http.StatusBadRequest, api.ErrorResponse(api.ErrUnparsable))
+		return
+	}
+
+	// Verify member role exists.
+	if params.Role == "" {
+		c.JSON(http.StatusBadRequest, api.ErrorResponse("member role is required"))
+		return
+	}
+
+	// Retrieve member from the database.
+	var member *db.Member
+	if member, err = db.RetrieveMember(c.Request.Context(), orgID, memberID); err != nil {
+		if errors.Is(err, db.ErrNotFound) {
+			c.JSON(http.StatusNotFound, api.ErrorResponse("member not found"))
+			return
+		}
+
+		sentry.Error(c).Err(err).Msg("could not retrieve member from the database")
+		c.JSON(http.StatusInternalServerError, api.ErrorResponse("could not update member role"))
+		return
+	}
+
+	// Check to ensure the memberID from the URL matches the member ID from the database.
+	if memberID != member.ID {
+		c.JSON(http.StatusBadRequest, api.ErrorResponse("member id does not match id in URL"))
+	}
+
+	// Update member role.
+	member.Role = params.Role
+
+	// Update member in the database.
+	if err = db.UpdateMember(c.Request.Context(), member); err != nil {
+		if errors.Is(err, db.ErrNotFound) {
+			c.JSON(http.StatusNotFound, api.ErrorResponse("member not found"))
+			return
+		}
+
+		sentry.Error(c).Err(err).Msg("could not update member in the database")
+		c.JSON(http.StatusInternalServerError, api.ErrorResponse("could not update member"))
+		return
+	}
+
+	// Get members from the database and set page size to return all members.
+	// TODO: Create list method that will not require pagination for this endpoint.
```

Comment:
I've created a story for this in the next sprint.

Reply:
`MemberList` will have pagination. A separate method, like what we previously had, will be created for `MemberRoleUpdate`,  `TenantStats` and `getOwner` since we don't really need to pass in a cursor for these read requests.

Reply:
Ah, that makes sense and seems like a good abstraction.